src/txn/txn_ckpt.c

```diff
@@ -1473,6 +1471,8 @@ __txn_checkpoint(WT_SESSION_IMPL *session, const char *cfg[])
     session->ckpt_handle_allocated = session->ckpt_handle_next = 0;
 
     session->isolation = txn->isolation = saved_isolation;
+    WT_STAT_CONN_SET(session, checkpoint_state, WT_CHECKPOINT_STATE_INACTIVE);
```

Comment:
What do you think about replacing this with another macro, `WT_CONN_STATE_SET`? It would check `WT_SESSION_IS_CHECKPOINT` and set the state, e.g.

```
#define WT_CONN_STATE_SET(session, state) \
    if (WT_SESSION_IS_CHECKPOINT((session)) \
        WT_STAT_CONN_SET((session), checkpoint_state, (state))
```

Reply:
We cannot have `WT_SESSION_IS_CHECKPOINT` inside the macro as it's not true before we have called `checkpoint_prepare` and it becomes false after finishing the checkpoint transaction.